src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -148,45 +148,17 @@ private function populateAvailableType($id, Definition $definition)
             $this->types[$type] = $id;
         }
 
-        // Cannot use reflection if the class isn't set
-        if (!$definition->getClass()) {
+        if (!$reflectionClass = $this->getReflectionClass($id, $definition)) {
             return;
         }
 
-        if ($reflectionClass = $this->getReflectionClass($id, $definition)) {
-            $this->extractInterfaces($id, $reflectionClass);
-            $this->extractAncestors($id, $reflectionClass);
-        }
-    }
-
-    /**
-     * Extracts the list of all interfaces implemented by a class.
-     *
-     * @param string           $id
-     * @param \ReflectionClass $reflectionClass
-     */
-    private function extractInterfaces($id, \ReflectionClass $reflectionClass)
-    {
-        foreach ($reflectionClass->getInterfaces() as $interfaceName => $reflectionInterface) {
-            $this->set($interfaceName, $id);
-
-            $this->extractInterfaces($id, $reflectionInterface);
```

Comment:
AFAIK, this is actually useless, as ``$reflectionClass->getInterfaces()`` gives you all implemented interfaces, even when they come from some inheritance. It does not give you only the interfaces appearing in the source code explicitly (but a check of the testsuite to ensure this case is covered should be necessary, to be sure that the code handles it right)